pkg/controller/clone-controller.go

```diff
@@ -158,7 +167,6 @@ func (cc *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 	//add the following annotation only if the pod pahse is succeeded, meaning job is completed
 	if phase == string(v1.PodSucceeded) {
 		anno[AnnCloneOf] = "true"
-		defer cc.deleteClonePods(sourcePod.Namespace, sourcePod.Name, targetPod.Name)
 	}
 	var lab map[string]string
```

Comment:
Separate issue, but I think it would also be wise to initialize `lab` as `lab := pvc.Labels`